src/Npgsql.NetTopologySuite/NpgsqlNetTopologySuiteExtensions.cs

```diff
@@ -0,0 +1,82 @@
+﻿#region License
+// The PostgreSQL License
+//
+// Copyright (C) 2018 The Npgsql Development Team
+//
+// Permission to use, copy, modify, and distribute this software and its
+// documentation for any purpose, without fee, and without a written
+// agreement is hereby granted, provided that the above copyright notice
+// and this paragraph and the following two paragraphs appear in all copies.
+//
+// IN NO EVENT SHALL THE NPGSQL DEVELOPMENT TEAM BE LIABLE TO ANY PARTY
+// FOR DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES,
+// INCLUDING LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS
+// DOCUMENTATION, EVEN IF THE NPGSQL DEVELOPMENT TEAM HAS BEEN ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+// THE NPGSQL DEVELOPMENT TEAM SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
+// ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
+// TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+#endregion
+
+using GeoAPI;
+using GeoAPI.Geometries;
+using NetTopologySuite.Geometries;
+using NetTopologySuite.IO;
+using Npgsql.NetTopologySuite;
+using Npgsql.TypeMapping;
+using NpgsqlTypes;
+using System.Data;
+
+namespace Npgsql
+{
+    /// <summary>
+    /// Extension allowing adding the NetTopologySuite plugin to an Npgsql type mapper.
+    /// </summary>
+    public static class NpgsqlNetTopologySuiteExtensions
+    {
+        /// <summary>
+        /// Sets up NetTopologySuite mappings for the PostGIS types.
+        /// </summary>
+        /// <param name="mapper">The type mapper to set up (global or connection-specific).</param>
+        /// <param name="coordinateSequenceFactory">The factory which knows how to build a particular implementation of ICoordinateSequence from an array of Coordinates.</param>
+        /// <param name="precisionModel">Specifies the grid of allowable points.</param>
+        /// <param name="handleOrdinates">Specifies the ordinates which will be handled. Not specified ordinates will be ignored.</param>
+        public static INpgsqlTypeMapper UseNetTopologySuite(
+            this INpgsqlTypeMapper mapper,
+            ICoordinateSequenceFactory coordinateSequenceFactory = null,
+            IPrecisionModel precisionModel = null,
+            Ordinates handleOrdinates = Ordinates.XYZM)
+        {
+            if (coordinateSequenceFactory == null)
+                coordinateSequenceFactory = GeometryServiceProvider.Instance.DefaultCoordinateSequenceFactory;
```

Comment:
`GeometryServiceProvider.Instance.DefaultCoordinateSequenceFactory` is -unless previously configured otherwise- CoordinateArraySequenceFactory. This factory cannot hande measure ordinate values.

You have to choose another one if you want to allow XYZM;

Reply:
Thanks! Which factory should be used instead? `DotSpatialAffineCoordinateSequenceFactory`?